src/OrchardCore.Modules/OrchardCore.Contents/Sitemaps/DefaultContentItemsQueryProvider.cs

```diff
@@ -10,50 +10,49 @@ namespace OrchardCore.Contents.Sitemaps;
 
 public class DefaultContentItemsQueryProvider : IContentItemsQueryProvider
 {
-    private readonly ISession _session;
+    private readonly IStore _store;
     private readonly IRouteableContentTypeCoordinator _routeableContentTypeCoordinator;
 
     public DefaultContentItemsQueryProvider(
-        ISession session,
-        IRouteableContentTypeCoordinator routeableContentTypeCoordinator
-        )
+        IStore store,
+        IRouteableContentTypeCoordinator routeableContentTypeCoordinator)
     {
-        _session = session;
+        _store = store;
         _routeableContentTypeCoordinator = routeableContentTypeCoordinator;
     }
 
-    public async Task GetContentItemsAsync(ContentTypesSitemapSource source, ContentItemsQueryContext context)
+    public async Task<ContentItemsQueryResult> GetContentItemsAsync(ContentTypesSitemapSource source, ContentItemsQueryContext context)
     {
+        ArgumentNullException.ThrowIfNull(source);
+        ArgumentNullException.ThrowIfNull(context);
+
         var routeableContentTypeDefinitions = await _routeableContentTypeCoordinator.ListRoutableTypeDefinitionsAsync();
 
+        var session = _store.CreateSession(withTracking: false);
```

Comment:
```suggestion
        using var session = _store.CreateSession(withTracking: false);
```

The session can be disposed after the method is called. And even without tracking it will then release the items.